CMakeLists.txt

```diff
@@ -12,6 +12,9 @@ add_subdirectory(3rdparty/catch2)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Woverloaded-virtual")
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_DEBUG_POSTFIX "d")
+
+set(CMAKE_BUILD_TYPE "Debug")
```

Comment:
This looks wrong. It prevents the user running cmake from defining the build type.

Reply:
resolve in commit 9344a35